reports/tasks.py

```diff
@@ -1,23 +1,68 @@
 from celery.task import Task
 from celery.utils.log import get_task_logger
+from celery.exceptions import SoftTimeLimitExceeded
 
+from django.core.exceptions import ObjectDoesNotExist
 
 logger = get_task_logger(__name__)
 
+from .models import Report
+from snappy.models import Message
 
-class The_Incr(Task):
+
+class Bounce_Report(Task):
 
     """
-    Task to incr something
+    Task to load and handle where reports need to go when complete
     """
-    name = "nightingale.reports.tasks.the_incr"
+    name = "reports.tasks.bounce_report"
+
+    class FailedEventRequest(Exception):
 
-    def run(self, anum, **kwargs):
         """
-        Returns an incr'd number
+        The attempted task failed because of a non-200 HTTP return
+        code.
+        """
+
+    def run(self, report_id, **kwargs):
+        """
+        Load and contruct message and send them off
         """
         l = self.get_logger(**kwargs)
-        l.info("Incrementing <%s>" % (anum,))
-        return int(anum)+1
 
-the_incr = The_Incr()
+        l.info("Loading Report")
+        try:
+            report = Report.objects.get(pk=report_id)
+            integrations = report.project.integrations
+            categories = report.categories.all()
+            active_snappy = integrations.filter(integration_type='Snappy',
+                                                active=True)
+            if active_snappy.count() == 1 and \
+                    "snappy_nonce" not in report.metadata:
+                # create a snappy message
+                content = "<b>Description:</b> %s <br>" % report.description
+                content += "<b>Categories:</b> <br>"
+                for category in categories:
+                    content += "%s <br>" % category.name
+                content += '<b>Location:</b> ' + \
+                    '<a href="https://www.google.co.za' +\
+                    '/maps/@%s,%s,13z">Map</a>' % (
+                        report.location.point.y, report.location.point.x)
+                message = Message()
+                message.integration = active_snappy[0]
+                message.report = report
+                message.target = "SNAPPY"
+                message.message = content
+                message.contact_key = report.contact_key
+                message.from_addr = report.to_addr
```

Comment:
In some places that becomes a to_addr in others it's a from_addr. But in the model context it's fixed.